test/Semantics/select-rank.f90

```diff
@@ -0,0 +1,266 @@
+
+   subroutine CALL_ME5(x)
+    implicit none
+    integer :: x(..),j
+    SELECT RANK(x)
+    RANK (0)
+      print *, "PRINT RANK 0"
+      j = INT(0, KIND=MERGE(KIND(0), -1, RANK(x) == 0))
+    RANK(1)
+      print *, "PRINT RANK 1"
+    !ERROR: Same rank values not allowed more than once
+    RANK(0)
+      print *, "ERROR"
+      j = INT(0, KIND=MERGE(KIND(0), -1, RANK(x) == 0))
+    END SELECT
+   end subroutine
+
+   subroutine CALL_ME6(x)
+    implicit none
+    integer :: x(..),j
+    SELECT RANK(x)
+    RANK (3)
+      print *, "one"
+      j = INT(0, KIND=MERGE(KIND(0), -1, RANK(x) == 3))
+    !ERROR: The value of the selector must be between zero and 15
+    RANK(-1)
+      print *, "rank: -ve"
+      j = INT(0, KIND=MERGE(KIND(0), -1, RANK(x) == -1))
+    END SELECT
+   end subroutine
+
+   subroutine CALL_ME7(arg)
+   implicit none
+   integer :: i,j
+   integer, dimension(..), pointer :: arg
+   integer, pointer :: arg2
+   !ERROR: RANK (*) cannot be used when selector is POINTER or ALLOCATABLE
+   select RANK(arg)
+   RANK (*)
+      print *, arg(1:1)
+   RANK (1)
+      print *, arg
+      j = INT(0, KIND=MERGE(KIND(0), -1, RANK(arg) == 1))
+   end select
+
+   !ERROR: Selector 'arg2' is not an assumed-rank array variable
+   select RANK(arg2)
+   RANK (*)
+      print *,"This would lead to crash when saveSelSymbol has std::nullptr"
+   RANK (1)
+      print *, "Rank is 1"
+   end select
+
+   end subroutine
+
+   subroutine CALL_ME8(x)
+    implicit none
+    integer :: x(..),j
+    SELECT RANK(x)
+    Rank(2)
+      print *, "Now it's rank 2 "
+    RANK (*)
+      print *, "Going for a other rank"
+      j = INT(0, KIND=MERGE(KIND(0), -1, RANK(x) == 1))
+    !ERROR: Not more than one of the selectors of SELECT RANK statement may be '*'
+    RANK (*)
+      print *, "This is Wrong"
+      j = INT(0, KIND=MERGE(KIND(0), -1, RANK(x) == 1))
+    END SELECT
+   end subroutine
+
+   subroutine CALL_ME10(x)
+    implicit none
+    integer:: x(..), a=10,b=20,j
+    integer, dimension(10) :: arr = (/1,2,3,4,5/),brr
+    integer :: const_variable=10
+    integer, pointer :: ptr,nullptr=>NULL()
+    type derived
+         character(len = 50) :: title
+    end type derived
+    type(derived) :: obj1
+
+    SELECT RANK(x)
+    Rank(2)
+      print *, "Now it's rank 2 "
+    RANK (*)
+      print *, "Going for a other rank"
+    !ERROR: Not more than one of the selectors of SELECT RANK statement may be '*'
+    RANK (*)
+      print *, "This is Wrong"
+    END SELECT
+
+    !ERROR: Selector 'brr' is not an assumed-rank array variable
+    SELECT RANK(ptr=>brr)
+    !ERROR: Must be a constant value
+    RANK(const_variable)
+      print *, "PRINT RANK 3"
+      j = INT(0, KIND=MERGE(KIND(0), -1, RANK(ptr) == 1))
+    !ERROR: Must be a constant value
+    RANK(nullptr)
+      print *, "PRINT RANK 3"
+    END SELECT
+
+    !ERROR: Selector 'x(1) + x(2)' is not an assumed-rank array variable
+    SELECT RANK (x(1) + x(2))
+    RANK(1)
```

Comment:
These `RANK` cases are not useful in these tests for errors on the `SELECT RANK` statement.